action/protocol/staking/handlers.go

```diff
@@ -663,10 +663,26 @@ func (p *Protocol) handleCandidateRegister(ctx context.Context, act *action.Cand
 		}
 	}
 
-	bucket := NewVoteBucket(owner, owner, act.Amount(), act.Duration(), blkCtx.BlockTimeStamp, act.AutoStake())
-	bucketIdx, err := csm.putBucketAndIndex(bucket)
-	if err != nil {
-		return log, nil, err
+	// register with self-stake
+	var (
+		bucketIdx = uint64(candidateNoSelfStakeBucketIndex)
+		txLogs    []*action.TransactionLog
+		votes     = big.NewInt(0)
+		err       error
+	)
+	if act.Amount().Sign() > 0 {
```

Comment:
no hardfork here?

Reply:
Before the hard fork, the `act.Amount` must be greater than 0, behaves the same as in the old version.

Reply:
`if act.Amount().Sign() <= 0 {` should be return error?

Reply:
already checked at https://github.com/iotexproject/iotex-core/pull/4059/files#diff-49d3555713cf037eb3ff80b28e13bd7329cc276b8fb0d60e4798b055a7e4aad8R70